### PR TITLE
[MSV1_0] dont change original pointers passed to LogonUserEx2

### PR DESCRIPTION
**Pull request by andy-123**
Old code makes lsass.exe unstable (tested on windows xp).

I discovered this bug during my work on sspi bringup branch.

### CODE REVIEW

File: dll/win32/msv1_0/msv1_0.c [hunk 1]
```diff
@@ -1275,20 +1285,18 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
...
+            LogonPass.Buffer = NULL;
+            LogonPass.MaximumLength = 0;
         }
 
-        TRACE("Domain: %S\n", LogonInfo->LogonDomainName.Buffer);
-        TRACE("User: %S\n", LogonInfo->UserName.Buffer);
-        TRACE("Password: %S\n", LogonInfo->Password.Buffer);
-
-        // TODO: If LogonType == Service, do some extra work using LogonInfo->Password.
+        TRACE("Domain: %.*S\n", LogonDom.Length / sizeof(WCHAR), LogonDom.Buffer);
```
Comment (ThFabba) [inline dll/win32/msv1_0/msv1_0.c]: Or even `%wZ` with `&LogonDom`

File: dll/win32/msv1_0/msv1_0.c [hunk 2]
```diff
@@ -1478,14 +1485,17 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
             }
 
             /* Check if the account expired */
-            AccountExpires.LowPart = UserInfo->All.AccountExpires.LowPart;
-            AccountExpires.HighPart = UserInfo->All.AccountExpires.HighPart;
-            if (LogonTime.QuadPart >= AccountExpires.QuadPart)
+            if (!(UserInfo->All.UserAccountControl & USER_DONT_EXPIRE_PASSWORD))
```
Comment (ThFabba) [inline dll/win32/msv1_0/msv1_0.c]: This seems like it belongs in a separate commit?

File: dll/win32/msv1_0/msv1_0.c (hunk 1 above)
Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: fixed

File: dll/win32/msv1_0/msv1_0.c (hunk 2 above)
Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: Reverted. Will be a new PR/commit.

File: dll/win32/msv1_0/msv1_0.c [hunk 3]
```diff
@@ -1275,20 +1285,18 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
...
-            LogonInfo->Password.MaximumLength = 0;
+            LogonPass.Buffer = NULL;
+            LogonPass.MaximumLength = 0;
         }
 
-        TRACE("Domain: %S\n", LogonInfo->LogonDomainName.Buffer);
-        TRACE("User: %S\n", LogonInfo->UserName.Buffer);
-        TRACE("Password: %S\n", LogonInfo->Password.Buffer);
-
-        // TODO: If LogonType == Service, do some extra work using LogonInfo->Password.
```
Comment (ThFabba) [inline dll/win32/msv1_0/msv1_0.c]: This TODO doesn't apply anymore?

File: dll/win32/msv1_0/msv1_0.c [hunk 4]
```diff
@@ -1244,29 +1247,36 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
         /* Fix-up pointers in the authentication info */
         PtrOffset = (ULONG_PTR)ProtocolSubmitBuffer - (ULONG_PTR)ClientBufferBase;
 
-        Status = RtlValidateUnicodeString(0, &LogonInfo->LogonDomainName);
+        /* We have to fix the string pointers.
+         * Changing the original pointer will make lsass unstable.
```
Comment (HBelusca) [inline dll/win32/msv1_0/msv1_0.c]: But do we know why / the actual reason that triggers the instability?

File: dll/win32/msv1_0/msv1_0.c (hunk 3 above)
Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: Sorry, my mistake. I restored this line.

File: dll/win32/msv1_0/msv1_0.c (hunk 4 above)
Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: No. After some investigation i think its not needed. Sorry.
This fix cames from sspi-bringup. I found some other things that i've fixed in msv1_0.dll. Maybe the pointer thing does only "improve" stability in any way but not "really" fix something.

If you wish i can close the PR or revert this change. If i revert the only change is to improve some TRACEs.

Comment (HBelusca) [reply dll/win32/msv1_0/msv1_0.c]: In the description of the PR you claim that it makes lsass unstable on XP (and 2003), so I don't know how you tested that (did you replace & use our msv1_0.dll on windows?)

Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: Yes, i've replaced msv1_0.dll on windows (xp and 2003). The patch came from my sspi-bringup. It was the first patch to LogonUserEx2. I've reverted this change im current sspi-bringup brach and no differnce (2003 works / xp failed with the same error as before).

Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: BTW: Current sspi-bringup msv1_0.dll is able too boot win2003+xp, win2003 can login (with and without password). And it seems to be stable.
